hydra-tx/src/Hydra/Tx/Snapshot.hs

```diff
@@ -37,6 +43,11 @@ fromChainSnapshotVersion :: Onchain.SnapshotVersion -> SnapshotVersion
 fromChainSnapshotVersion =
   UnsafeSnapshotVersion . fromMaybe 0 . integerToNatural
 
+instance Arbitrary SnapshotVersion where
+  arbitrary = UnsafeSnapshotVersion <$> arbitrary
```

Comment:
Can these not be `deriving anyclass Arbitrary` ?

Reply:
Unfortunately not, because there is [no default methods](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/derive_any_class.html) on [Arbitrary](https://hackage.haskell.org/package/QuickCheck-2.15.0.1/docs/Test-QuickCheck.html#t:Arbitrary).

But we can use `deriving newtype`, which I did here now: https://github.com/cardano-scaling/hydra/tree/3ab8034cc196892e117972544c831f620a252877/./